unraid_mcp/core/guards.py

```
@@ -0,0 +1,101 @@
+            Pass a str when one description covers all destructive actions.
+            Pass a dict[action_name, description] when descriptions differ.
+    """
+    if action not in destructive_actions or confirm:
+        return
+
+    desc = description[action] if isinstance(description, dict) else description
+    confirmed = await elicit_destructive_confirmation(ctx, action, desc)
+    if not confirmed:
+        raise ToolError(
+            f"Action '{action}' was not confirmed. Re-run with confirm=True to bypass elicitation."
+        )
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)